samples/third-party/gltf-loader.js

```diff
@@ -0,0 +1,354 @@
+                                        , data[stride * i + offset + 1]
+                                        , data[stride * i + offset + 2]
+                                        , 0);
+                        vec4.transformMat4(tmpVec4, tmpVec4, inverseTransposeMatrix);
+                        vec4.normalize(tmpVec4, tmpVec4);
+                        data[stride * i + offset] = tmpVec4[0];
+                        data[stride * i + offset + 1] = tmpVec4[1];
+                        data[stride * i + offset + 2] = tmpVec4[2];
+                    }
+                }
+            }
+            
+            if (data) {
+                scene.vertices = data;
+
+                // Completed loading
+                callback(scene);
+            }
+        });
+    };
+
+    window.loadGltfGeometry = function(url, onload) {
+        
+        // @todo: currently using a relative path to load gl-matrix-min
+        (function(src, callback){
+            // http://stackoverflow.com/questions/950087/include-a-javascript-file-in-another-javascript-file
+            
+            // Adding the script tag to the head as suggested before
+            var head = document.getElementsByTagName('head')[0];
+            var script = document.createElement('script');
+            script.type = 'text/javascript';
+            script.src = src;
+
+            // Then bind the event to the callback function.
+            // There are several events for cross browser compatibility.
+            script.onreadystatechange = callback;
+            script.onload = callback;
+
+            // Fire the loading
+            head.appendChild(script);
+        })('third-party/gl-matrix-min.js', 
+            (function() {
+            // Save path to load .bin file
+            GLTF.baseUri = getBaseUri(url);
+            
+            // matrix transform preparation
+            translationVec3 = vec3.create();
+            rotationQuat = vec4.create();
+            scaleVec3 = vec3.create();
+            TRMatrix = mat4.create();
+            
+            tmpVec4 = vec4.create();
+            inverseTransposeMatrix = mat4.create();
+
+            loadJSON(url, function(response) {
+                // Parse JSON string into object
+                var jsonObj = JSON.parse(response);
+                parseGltf(jsonObj, onload);
+            });
+        })); 
+ 
+    };
+    
+    function getBaseUri(uri) {
+        
+        // https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/getBaseUri.js
+        
+        var basePath = '';
+        var i = uri.lastIndexOf('/');
+        if(i !== -1) {
+            basePath = uri.substring(0, i + 1);
+        }
+        
+        return basePath;
+    }
+
+    function loadJSON(src, callback) {
+
+        // native json loading technique from @KryptoniteDove:
+        // http://codepen.io/KryptoniteDove/post/load-json-file-locally-using-pure-javascript
```

Comment:
The reduce the number of places this includes third-party code from, you might want to base this and `loadArrayBuffer` off of the versions in Cesium.